internal/manifests/mutate.go

```diff
@@ -224,12 +224,18 @@ func mutateRole(existing, desired *rbacv1.Role) {
 	existing.Annotations = desired.Annotations
 	existing.Labels = desired.Labels
 	existing.Rules = desired.Rules
+	// This role can exists in a different namespace than the otel collector, so we need to remove the owner references
+	// since cross namespace owner references are not allowed
+	existing.SetOwnerReferences(nil)
```

Comment:
why does the role not exist in the same namespace?

Reply:
this isn't a cluster role right?

Reply:
The thing is that the resource can be created by an OpenTelemetry Collector from another namespace https://github.com/open-telemetry/opentelemetry-operator/pull/3459/files#r1865462426